kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/serialization/AxonSerializers.kt

```diff
@@ -0,0 +1,346 @@
+        subclass(ConfigTokenSerializer)
+        subclass(GapAwareTrackingTokenSerializer)
+        subclass(MultiSourceTrackingTokenSerializer)
+        subclass(MergedTrackingTokenSerializer)
+        subclass(ReplayTokenSerializer)
+        subclass(GlobalSequenceTrackingTokenSerializer)
+    }
+
+    contextual(SimpleScheduleToken::class) { SimpleScheduleTokenSerializer }
+    contextual(QuartzScheduleToken::class) { QuartzScheduleTokenSerializer }
+    polymorphic(ScheduleToken::class) {
+        subclass(SimpleScheduleTokenSerializer)
+        subclass(QuartzScheduleTokenSerializer)
+    }
+
+    contextual(InstanceResponseType::class) { InstanceResponseTypeSerializer }
+    contextual(OptionalResponseType::class) { OptionalResponseTypeSerializer }
+    contextual(MultipleInstancesResponseType::class) { MultipleInstancesResponseTypeSerializer }
+    contextual(ArrayResponseType::class) { ArrayResponseTypeSerializer }
+    polymorphic(ResponseType::class) {
+        subclass(InstanceResponseTypeSerializer)
+        subclass(OptionalResponseTypeSerializer)
+        subclass(MultipleInstancesResponseTypeSerializer)
+        subclass(ArrayResponseTypeSerializer)
+    }
+}
+
+object ConfigTokenSerializer : KSerializer<ConfigToken> {
+
+    private val mapSerializer = MapSerializer(String.serializer(), String.serializer())
+    override val descriptor = buildClassSerialDescriptor(ConfigToken::class.java.name) {
+        element<Map<String, String>>("config")
+    }
+
+    override fun deserialize(decoder: Decoder): ConfigToken = decoder.decodeStructure(descriptor) {
+        var config: Map<String, String>? = null
+        while (true) {
+            val index = decodeElementIndex(descriptor)
+            if (index == CompositeDecoder.DECODE_DONE) break
+            when (index) {
+                0 -> config = decodeSerializableElement(descriptor, index, mapSerializer)
+            }
+        }
+        ConfigToken(
+            config ?: throw SerializationException("Element 'config' is missing"),
+        )
+    }
+
+    override fun serialize(encoder: Encoder, value: ConfigToken) = encoder.encodeStructure(descriptor) {
+        encodeSerializableElement(descriptor, 0, mapSerializer, value.config)
+    }
+}
+
+object GapAwareTrackingTokenSerializer : KSerializer<GapAwareTrackingToken> {
+
+    private val setSerializer = SetSerializer(Long.serializer())
+    override val descriptor = buildClassSerialDescriptor(GapAwareTrackingToken::class.java.name) {
+        element<Long>("index")
+        element("gaps", setSerializer.descriptor)
+    }
+
+    override fun deserialize(decoder: Decoder) = decoder.decodeStructure(descriptor) {
+        var gapIndex: Long? = null
+        var gaps: Set<Long>? = null
+        while (true) {
+            val index = decodeElementIndex(descriptor)
+            if (index == CompositeDecoder.DECODE_DONE) break
+            when (index) {
+                0 -> gapIndex = decodeLongElement(descriptor, index)
```

Comment:
Question, I assume the values 0 and 1 reflecting the fields `index` and `gaps` mean that if we'd reorder the fields in the `GapAwareTrackingToken`, this would break the serialization?